### PR TITLE
KTEST: Support 'posixOnly' attributes in config.xml

### PR DESCRIPTION
**Pull request by osa1**
This is modified version of @daejunpark 's patch. I made changes that we discussed at original pull request.

@dwightguth please tell me if I'm forgetting anything.

### CODE REVIEW

**Comment by dwightguth** [comment]
add to whitelist

File: lib/ktest.xsd [hunk 1]
```diff
@@ -48,6 +48,7 @@
           <xs:attribute name="exclude" type="xs:string" use="optional" />
           <xs:attribute name="results" type="xs:string" use="optional" />
           <xs:attribute name="skip" type="xs:string" use="optional" />
+          <xs:attribute name="posixOnly" type="xs:string" use="optional" />
```
Comment (dwightguth) [inline lib/ktest.xsd]: shouldn't this be called posixInitScript also?

File: src/javasources/KTool/src/org/kframework/ktest/Test/TestCase.java [hunk 2]
```diff
@@ -95,7 +102,7 @@ public static TestCase makeTestCaseFromK(CmdArg cmdArgs) throws InvalidConfigErr
 
         return new TestCase(targetFile, programs, cmdArgs.getExtensions(),
                 cmdArgs.getExcludes(), results, emptyOpts, emptyOpts, emptyOptsMap,
-                new HashSet<KTestStep>());
+                new HashSet<KTestStep>(), "");
```
Comment (dwightguth) [inline src/javasources/KTool/src/org/kframework/ktest/Test/TestCase.java]: Either make this set via a setter or else create a separate constructor in which this parameter is not present.

File: src/javasources/KTool/src/org/kframework/ktest/Test/TestSuite.java [hunk 3]
```diff
@@ -159,6 +170,46 @@ public void dryRun() {
...
+        assert GlobalSettings.isPosix();
+        int len = tests.size();
+        List<TestCase> successfulTests = new ArrayList<>(len);
+        List<Proc<TestCase>> ps = new ArrayList<>(len);
+
+        System.out.format("Running initial scripts...%n");
+        startTpe();
+        for (TestCase tc : tests) {
+            if (tc.hasPosixOnly()) {
+                Proc<TestCase> p = new Proc<>(tc, tc.getPosixOnlyCmd(), tc.toPosixOnlyLogString(),
```
Comment (dwightguth) [inline src/javasources/KTool/src/org/kframework/ktest/Test/TestSuite.java:188]: I'm still not comfortable with all these parameters being passed along and then not used. We don't have to do the complete refactoring now if you don't want, but at the very least it should be possible to create processes without specifying information that has nothing to do with the conceptual notion of a process.

**Comment by dwightguth** [comment]
Fix these last couple things and then we can merge it if the tests pass.

**Comment by osa1** [comment]
Hmm, for some reason I can't run K on my system anymore.

```
➜  k git:(daejun-patch) ✗ ./bin/ktest 
Error: Could not find or load main class org.kframework.main.Main
➜  k git:(daejun-patch) ✗ java -jar lib/java/k3.jar      
Error: Could not find or load main class org.kframework.main.Main
```

File: src/javasources/KTool/src/org/kframework/ktest/Test/TestSuite.java (hunk 3 above)
Comment (osa1) [inline src/javasources/KTool/src/org/kframework/ktest/Test/TestSuite.java:188]: I deliberately left refactoring to it's own pull request.

Comment (dwightguth) [inline src/javasources/KTool/src/org/kframework/ktest/Test/TestSuite.java:188]: Okay, then create an issue for this and assign it to yourself so I can track it. This needs to happen without too much unnecessary delay.

**Comment by daejunpark** [comment]
@osa1 @dwightguth I'm really sorry not to respond properly, and really thank you Omer for resolving all of these. Many thanks again, Omer!

**Pull request merged**